upstairs/src/volume.rs

```diff
@@ -611,6 +611,36 @@ impl Volume {
 
         Ok(())
     }
+
+    pub async fn volume_extent_info(
+        &self,
+    ) -> Result<VolumeInfo, CrucibleError> {
+        // A volume has multiple levels of extent info, not just one.
+        let mut volumes = Vec::new();
+
+        for sub_volume in &self.sub_volumes {
+            match sub_volume.query_extent_info().await? {
+                Some(ei) => {
+                    assert_eq!(self.block_size, ei.block_size);
+                    let svi = SubVolumeInfo {
+                        blocks_per_extent: ei.blocks_per_extent,
+                        extent_count: ei.extent_count,
+                    };
+                    volumes.push(svi);
+                }
+                None => {
+                    continue;
```

Comment:
I would lean towards "mixing extent and non-extent flavored subvolumes is an error", instead of ignoring it.

(In other words, return an error if we get a `None` for a subvolume and `!volumes.is_empty()`)

Reply:
I've updated this to return an error if the subvolumes don't match.